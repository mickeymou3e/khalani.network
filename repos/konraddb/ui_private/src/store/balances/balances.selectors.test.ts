import {
  customerCodeMock,
  jltCustodyWalletCode,
  jltWithMarketPriceMock,
  maticPolygonCustodyWalletCode,
  tradedBalancesMock,
} from "@/definitions/__mocks__";

import {
  selectBalancesList,
  selectCryptoBalanceAssets,
  selectCryptoBalancesList,
  selectFiatBalanceAssets,
  selectFiatBalancesList,
} from "./balances.selectors";

describe("Balances selectors and utils", () => {
  describe("selectBalancesList", () => {
    it("should return a list of balances with formatted values", () => {
      const result = selectBalancesList.resultFunc(
        [maticPolygonCustodyWalletCode, jltCustodyWalletCode],
        customerCodeMock,
        tradedBalancesMock,
        [jltWithMarketPriceMock]
      );

      expect(result).toMatchSnapshot();
    });
  });

  describe("selectCryptoBalancesList", () => {
    it("should return a list of crypto balances", () => {
      const intermediateResult = selectBalancesList.resultFunc(
        [maticPolygonCustodyWalletCode, jltCustodyWalletCode],
        customerCodeMock,
        tradedBalancesMock,
        [jltWithMarketPriceMock]
      );
      const result = selectCryptoBalancesList.resultFunc(intermediateResult);

      expect(result).toMatchSnapshot();
    });
  });

  describe("selectFiatBalancesList", () => {
    it("should return a list of fiat balances", () => {
      const intermediateResult = selectBalancesList.resultFunc(
        [maticPolygonCustodyWalletCode, jltCustodyWalletCode],
        customerCodeMock,
        tradedBalancesMock,
        [jltWithMarketPriceMock]
      );
      const result = selectFiatBalancesList.resultFunc(intermediateResult);

      expect(result).toMatchSnapshot();
    });
  });

  describe("selectCryptoBalanceAssets", () => {
    it("should return a list of balance assets without fiat currencies", () => {
      const result = selectCryptoBalanceAssets.resultFunc(tradedBalancesMock);
      const expected = [
        {
          value: "JLT-F23",
          assets: [
            {
              icon: "JLT-F23",
              label: "JLT-F23",
              description: "JLT-F23",
            },
          ],
        },
      ];

      expect(result).toEqual(expected);
    });
  });

  describe("selectFiatBalanceAssets", () => {
    it("should return a list of balance assets with fiat currencies", () => {
      const result = selectFiatBalanceAssets.resultFunc(tradedBalancesMock);
      const expected = [
        {
          value: "EUR",
          assets: [
            {
              icon: "EUR",
              label: "EUR",
              description: "Euro",
            },
          ],
        },
      ];

      expect(result).toEqual(expected);
    });
  });
});
