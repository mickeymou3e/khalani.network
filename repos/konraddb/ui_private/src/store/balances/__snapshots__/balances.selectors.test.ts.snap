// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Balances selectors and utils selectBalancesList should return a list of balances with formatted values 1`] = `
[
  {
    "asset": "JLT-F23",
    "assetName": "JLT-F23",
    "base": {
      "availableToAux": "200.00",
      "availableToAuxValue": 200,
      "availableToTrade": "250.00",
      "availableToTradeValue": 250,
      "inOrders": "0.00",
      "inOrdersValue": 0,
      "locked": "50.00",
      "lockedValue": 50,
      "total": "250.00",
      "totalValue": 250,
    },
    "description": "Voluntary REC Front - Half 2023",
    "icon": "JLT-F23",
    "id": "JLT-F23",
    "isFiat": false,
    "quote": {
      "availableToAux": "408.11 EUR",
      "availableToAuxValue": 408.1075655277388,
      "availableToTrade": "510.13 EUR",
      "availableToTradeValue": 510.1344569096735,
      "inOrders": "0.00 EUR",
      "inOrdersValue": 0,
      "locked": "102.03 EUR",
      "lockedValue": 102.0268913819347,
      "total": "510.13 EUR",
      "totalValue": 510.1344569096735,
    },
  },
  {
    "asset": "EUR",
    "assetName": "Euro",
    "base": {
      "availableToAux": "1,000.25",
      "availableToAuxValue": 1000.25,
      "availableToTrade": "1,000.25",
      "availableToTradeValue": 1000.25,
      "inOrders": "10.00",
      "inOrdersValue": 10,
      "locked": "0.00",
      "lockedValue": 0,
      "total": "1,010.25",
      "totalValue": 1010.25,
    },
    "description": "Euro",
    "icon": "EUR",
    "id": "EUR",
    "isFiat": true,
    "quote": {
      "availableToAux": "1,000.25 EUR",
      "availableToAuxValue": 1000.25,
      "availableToTrade": "1,000.25 EUR",
      "availableToTradeValue": 1000.25,
      "inOrders": "10.00 EUR",
      "inOrdersValue": 10,
      "locked": "0.00 EUR",
      "lockedValue": 0,
      "total": "1,010.25 EUR",
      "totalValue": 1010.25,
    },
  },
]
`;

exports[`Balances selectors and utils selectCryptoBalancesList should return a list of crypto balances 1`] = `
[
  {
    "asset": "JLT-F23",
    "assetName": "JLT-F23",
    "base": {
      "availableToAux": "200.00",
      "availableToAuxValue": 200,
      "availableToTrade": "250.00",
      "availableToTradeValue": 250,
      "inOrders": "0.00",
      "inOrdersValue": 0,
      "locked": "50.00",
      "lockedValue": 50,
      "total": "250.00",
      "totalValue": 250,
    },
    "description": "Voluntary REC Front - Half 2023",
    "icon": "JLT-F23",
    "id": "JLT-F23",
    "isFiat": false,
    "quote": {
      "availableToAux": "408.11 EUR",
      "availableToAuxValue": 408.1075655277388,
      "availableToTrade": "510.13 EUR",
      "availableToTradeValue": 510.1344569096735,
      "inOrders": "0.00 EUR",
      "inOrdersValue": 0,
      "locked": "102.03 EUR",
      "lockedValue": 102.0268913819347,
      "total": "510.13 EUR",
      "totalValue": 510.1344569096735,
    },
  },
]
`;

exports[`Balances selectors and utils selectFiatBalancesList should return a list of fiat balances 1`] = `
[
  {
    "asset": "EUR",
    "assetName": "Euro",
    "base": {
      "availableToAux": "1,000.25",
      "availableToAuxValue": 1000.25,
      "availableToTrade": "1,000.25",
      "availableToTradeValue": 1000.25,
      "inOrders": "10.00",
      "inOrdersValue": 10,
      "locked": "0.00",
      "lockedValue": 0,
      "total": "1,010.25",
      "totalValue": 1010.25,
    },
    "description": "Euro",
    "icon": "EUR",
    "id": "EUR",
    "isFiat": true,
    "quote": {
      "availableToAux": "1,000.25 EUR",
      "availableToAuxValue": 1000.25,
      "availableToTrade": "1,000.25 EUR",
      "availableToTradeValue": 1000.25,
      "inOrders": "10.00 EUR",
      "inOrdersValue": 10,
      "locked": "0.00 EUR",
      "lockedValue": 0,
      "total": "1,010.25 EUR",
      "totalValue": 1010.25,
    },
  },
]
`;
